### PR TITLE
[Messenger] document reset_on_message transport option

### PR DESCRIPTION
**Pull request by lyrixx**
refs https://github.com/symfony/symfony/pull/41163

### CODE REVIEW

File: messenger.rst [hunk 1]
```diff
@@ -696,6 +696,68 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
     [program:x]
     stopwaitsecs=20
 
+
+Stateless worker
+~~~~~~~~~~~~~~~~
+
+PHP was designed to be stateless: everything is lost after processing an HTTP
+request. When you run your application in a HTTP context, you don't have to
+clean your services since PHP clean everything after sending the request.
```
Comment (94noni) [inline messenger.rst]: Sending the response instead

Also there are some services a dev can clean inside the web flow like the entity manager in doctrine context etc

Thus I would replace « don’t have to clean » to « you may not take care of services states that may leak »

File: messenger.rst [hunk 2]
```diff
@@ -696,6 +696,68 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
     [program:x]
     stopwaitsecs=20
 
+
+Stateless worker
+~~~~~~~~~~~~~~~~
+
+PHP was designed to be stateless: everything is lost after processing an HTTP
```
Comment (94noni) [inline messenger.rst:703]: « was » or « is » ?

File: messenger.rst [hunk 3]
```diff
@@ -696,6 +696,68 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
...
+request. When you run your application in a HTTP context, you don't have to
+clean your services since PHP clean everything after sending the request.
+
+Since worker run in a CLI context, you need to be careful about services state.
+You should avoid to put a state in a service to avoid leaking some information
```
Comment (94noni) [inline messenger.rst:709]: What about « information and/or memory »

File: messenger.rst (hunk 2 above)
Comment (OskarStark) [reply messenger.rst:703]: Is

File: messenger.rst [hunk 4]
```diff
@@ -696,6 +696,68 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
     [program:x]
     stopwaitsecs=20
 
+
+Stateless worker
+~~~~~~~~~~~~~~~~
+
+PHP was designed to be stateless: everything is lost after processing an HTTP
+request. When you run your application in a HTTP context, you don't have to
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
request. When you run your application in an HTTP context, you don't have to
```

File: messenger.rst [hunk 5]
```diff
@@ -696,6 +696,68 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
...
+clean your services since PHP clean everything after sending the request.
+
+Since worker run in a CLI context, you need to be careful about services state.
+You should avoid to put a state in a service to avoid leaking some information
+from one message to another message.
+
+Some symfony services leak by nature. For example the monolog fingers cross
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
Some symfony services leak by nature. For example the monolog fingers crossed
```

**Comment by lyrixx** [comment]
@OskarStark & @94noni thanks for the review. I have addressed your comments.

File: messenger.rst [hunk 6]
```diff
@@ -696,6 +696,69 @@ of the desired grace period in seconds) in order to perform a graceful shutdown:
     [program:x]
     stopwaitsecs=20
 
+
+Stateless worker
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
Stateless Worker
```

**Comment by javiereguiluz** [comment]
Thanks Grég for this nice contribution! I did some minor rewords while merging, but the core of your contribution remained. Cheers!

**Pull request merged**